extensions/wikia/UserLogin/UserLoginSpecialController.class.php

```diff
@@ -74,33 +54,53 @@ private function initializeTemplate() {
 		$this->getOutput()->disallowUserJs(); // just in case...
 	}
 
+	private function isMonobookOrUncyclo() {
+		$skin = $this->wg->User->getSkin();
+		return ( $skin instanceof SkinMonoBook || $skin instanceof SkinUncyclopedia );
+	}
+
 	/**
 	 * Route the view based on logged in status
 	 */
 	public function index() {
-		$contentLangCode = $this->app->wg->ContLang->getCode();
-
-		// Redirect to standalone NewAuth page if extension enabled
-		if ( $this->app->wg->EnableNewAuthModal ) {
-			$redirectUrl = $this->getAuthenticationResource()
-				. '?redirect=' . $this->userLoginHelper->getRedirectUrl();
-
-			if ( $contentLangCode !== 'en' ) {
-				$redirectUrl .= '&uselang=' . $contentLangCode;
-			}
-
-			$this->getOutput()->redirect( $redirectUrl );
-		}
-
 		if ( $this->wg->User->isLoggedIn() ) {
 			$this->forward( __CLASS__, 'loggedIn' );
 		} else {
 			$this->forward( __CLASS__, 'loginForm' );
 		}
 	}
 
+	private function redirectToMercury() {
+		$out = $this->getOutput();
+		$out->redirect( $this->getRedirectUrl() );
+		$this->clearBodyAndSetMaxCache( $out );
```

Comment:
Pending https://github.com/Wikia/app/pull/12817 this code can be extracted to a more common place.